.github/workflows/publish.yml

```diff
@@ -0,0 +1,139 @@
+name: Publish
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  contracts:
+    runs-on: token-studio-linux-medium
+    container:
+      image: node:20.17.0-alpine3.20 # Using alpine for a smaller image
+    permissions:
+      contents: read
+
+    steps:
+      # * Initial steps
+      - name: Checkout repository
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
+
+      - name: Initial steps
+        uses: ./.github/actions/initial-steps
+        with:
+          deps-contracts-install: "true"
+          deps-change-references: "true"
+
+      - name: Create file .npmrc
+        run: |
+          cat <<EOF > .npmrc
+          //registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}
+          # //localhost:4873/:_authToken=${{ secrets.NPM_TOKEN }} # ! Only for local testing, DO NOT uncomment for production
+          EOF
+        working-directory: contracts
+
+      # # ! Only for local testing, DO NOT uncomment for production
+      # - name: Set NPM registry to Verdaccio (local)
+      #   run: npm config set registry http://localhost:4873
+
+      - name: Change references to repo
+        run: ./changeProyectsReferencesToRepo.sh
+
+      - name: Install Contracts
+        run: npm ci
+        working-directory: contracts
+
+      - name: Publish contracts
+        run: npm run publish:contracts --access=public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  sdk:
+    # needs: contracts
+    runs-on: token-studio-linux-medium
+    container:
+      image: node:20.17.0-alpine3.20 # Using alpine for a smaller image
+    permissions:
+      contents: read
+
+    steps:
+      # * Initial steps
+      - name: Checkout repository
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
```

Comment:
Add specific version as # after the commit hash.

Reply:
version added